extensions/iam/oauth2/oauth2-core/src/main/java/org/eclipse/dataspaceconnector/iam/oauth2/core/impl/Oauth2ServiceImpl.java

```diff
@@ -66,32 +67,37 @@
     private final List<ValidationRule> validationRules;
     private final JWSSigner tokenSigner;
     private final JwtDecoratorRegistry jwtDecoratorRegistry;
+    private final JWSHeader.Builder jwsHeaderBuilder;
 
     /**
      * Creates a new instance of the OAuth2 Service
      *
      * @param configuration             The configuration
-     * @param signerProvider            A {@link Supplier} which is used to get a {@link JWSSigner} instance.
+     * @param tokenSigner               A {@link JWSSigner} instance.
      * @param client                    Http client
      * @param jwtDecoratorRegistry      Registry containing the decorator for build the JWT
      * @param typeManager               Type manager
      * @param additionalValidationRules An optional list of {@link ValidationRule} that are evaluated <em>after</em> the
      *                                  standard OAuth2 validation
      */
-    public Oauth2ServiceImpl(Oauth2Configuration configuration, Supplier<JWSSigner> signerProvider, OkHttpClient client, JwtDecoratorRegistry jwtDecoratorRegistry, TypeManager typeManager, ValidationRule... additionalValidationRules) {
+    public Oauth2ServiceImpl(Oauth2Configuration configuration, JWSSigner tokenSigner, OkHttpClient client, JwtDecoratorRegistry jwtDecoratorRegistry, TypeManager typeManager, ValidationRule... additionalValidationRules) {
         this.configuration = configuration;
         this.typeManager = typeManager;
         httpClient = client;
         this.jwtDecoratorRegistry = jwtDecoratorRegistry;
+        this.tokenSigner = tokenSigner;
 
         List<ValidationRule> rules = new ArrayList<>();
         rules.add(new Oauth2ValidationRule()); //OAuth2 validation must ALWAYS be done
         rules.addAll(List.of(additionalValidationRules));
         validationRules = Collections.unmodifiableList(rules);
 
-        tokenSigner = signerProvider.get();
-        if (tokenSigner == null) {
-            throw new EdcException("Could not resolve private key");
+        if (tokenSigner instanceof ECDSASigner) {
+            //supports ECDSA private key
+            jwsHeaderBuilder = new JWSHeader.Builder(JWSAlgorithm.ES256);
```

Comment:
wait, this is buggy, it the algorithm that should be set as field, but the builder should be recreated every time a JWT needs to be build, otherwise it could lead to problems

Reply:
@tmberthold can you please take a look?

Reply:
already fixed with new PR: https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/pull/481